core/player/src/player.ts

```diff
@@ -87,12 +87,41 @@ export class Player {
     commit: COMMIT,
   };
 
-  public readonly logger = new TapableLogger();
-  public readonly constantsController = new ConstantsController();
+  public readonly logger: TapableLogger = new TapableLogger();
+  public readonly constantsController: ConstantsController =
+    new ConstantsController();
   private config: PlayerConfigOptions;
   private state: PlayerFlowState = NOT_STARTED_STATE;
 
-  public readonly hooks = {
+  public readonly hooks: {
```

Comment:
Any way to skip the verboseness for this one? I worry someone won't update both and we'll have a weird type error.

Reply:
If they don't update both it would fail TS type checking which, IMO, is a good thing.